### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Ewoskyx**
### Add forms

In this PR I did;

Create  Forms:
- Create a Post on behalf of the current_user.
- Creat a comment on behalf of the current_user.

Allow Users to add likes to Posts

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+    )
+    @new_comment.post_id = @post.id
+    @new_comment.update_comments_counter
+    if @new_comment.save
+      redirect_to "/users/#{@post.author_id}/posts/#{@post.id}"
+    else
+      render text: 'Could not save post'
+    end
+  end
+
```
Comment (andersonlebon) [inline app/controllers/comments_controller.rb]: - [x] kindly fix this issue that is rendering the complete hash in the view. you can fix it by extracting the text in the params when creating. Hint :point_down: 

```rb
    @new_comment = @user.comments.new(
      text: comment_params[:text],
      author_id: @user.id,
      post_id: @post.id
    )
```


image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,2 +1,28 @@
 class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @user = User.find(1)
```
Comment (andersonlebon) [inline app/controllers/comments_controller.rb]: - [x] Note that the comment should not only be created for the user with the `id: 1` instead it should be created using the id of the current user which is in the params. This current user can be obtained by setting it correctly the routes. example :point_down: 

```rb
Rails.application.routes.draw do
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show: :new, :create] do
      resources :comments, only: %i[create new update destroy]
      resources :likes, only: %i[create]
    end 
  end

  # Defines the root path route ("/")
  root "users#index"
end

**Review by andersonlebon — COMMENTED** [review]
## Hi @Ewoskyx 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @Ewoskyx, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**